### DISCUSSION TITLE
suggestion to add space and dot separated BIBLE chapter and verse references and a new prefix trigger

### DISCUSSION THREAD

**Discussion by seesmof** [tim-hub/obsidian-bible-reference#223] [Ideas]
GOD bless you guys

suggestions are:
1. add a dot or space separated BIBLE chapter and verse numbers
2. add a longer dash as caller for better experience
3. add a space after inserting a Passage using Verse Lookup feature

### 1. making reference tweaks

its convenient to write BIBLE references with dot or space separated verse and chapter numbers
_for example:_ `JN3 16` - BlueLetterBIBLE would accept that, or `JN14.6` - BIBLEgateway would accept that

code changes would be in `src/utils/regs.ts` as follows:
```js
export const MODAL_REG = /([123])*\s*[A-z]{2,}\s*\d{1,3}[:. ]\d{1,3}(-\d{1,3})*/
```

### 2. adding a new Passage caller

in obsidian there's a useful plugin called 'Smart Typography' which turns a `--` into `–` while typing
it would be good to allow this symbol as a trigger also

can it hinder performance in any way? 
is it a good idea to add such a trigger?
can we do that?

the changes would also be in `src/utils/regs.ts` like that:
```js
export const DEFAULT_TRIGGER_PREFIX_REG = /--|(\+\+)|–/
```

### 3. adding space after verse lookup insertion

when using the Verse Lookup Modal feature i noticed that it inserts the BIBLE Passage at cursor position
once it does, the cursor is no longer on the last line in the file and it's not too convenient to continue writing like that
when using just the caller and inserting a Passage inline like that, all works well: the cursor gets placed after the BIBLE block

how to accomplish the same thing in the Verse Lookup feature i do not know
but i think these code blocks are responsible for insertion, am i right?:

#### inline by trigger

`src/suggesetor/VerseEditorSuggester.ts` lines `101-110`:
```js
selectSuggestion(suggestion: VerseSuggesting): void {
  if (this.context) {
    /* prettier-ignore */
    (this.context.editor as Editor).replaceRange(
      suggestion.allFormattedContent,
      this.context.start,
      this.context.end,
    )
  }
}
```

#### from modal

`src/suggesetor/VerseLookupSuggestModal.ts` lines `48-54`:
```js
onChooseSuggestion(item: VerseSuggesting, evt: MouseEvent | KeyboardEvent) {
  const editor = this.app.workspace.getActiveViewOfType(MarkdownView)?.editor
  if (!editor) {
    return
  }
  editor.replaceRange(item.allFormattedContent, editor.getCursor())
}
```

one idea is to insert a cursor after the BIBLE Passage from modal
JESUS helped us find this:
```js
onChooseSuggestion(item: VerseSuggesting, evt: MouseEvent | KeyboardEvent) {
  const editor = this.app.workspace.getActiveViewOfType(MarkdownView)?.editor;
  if (!editor) {
      return;
  }

  const cursor = editor.getCursor();
  editor.replaceRange(item.allFormattedContent, cursor);
  const newCursorPosition = { line: cursor.line, ch: cursor.ch + item.allFormattedContent.length };
  editor.setCursor(newCursorPosition);
}
```
will this work?

GOD bless you guys
may JESUS CHRIST our HOLY LORD GOD ALMIGHTY KING OF KINGS AND LORD OF LORDS bless you greatly and establish all the works of your hands!

**Comment by tim-hub** [comment]
Thanks for your ideas, I like them, and let me find time to implement them.
